### PR TITLE
Move the NAT Gateways to the public subnets

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

I had previously put the NAT Gateways in the private subnets.  This is foolish, since then they cannot reach out to the internet!  This pull request fixes my bone-headed mistake.

## 💭 Motivation and Context

The private subnets cannot reach out to the internet without this fix, so it is very necessary.

## 🧪 Testing

I deployed these changes to production and verified that the private subnets can now reach out to the internet.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: subnets.tf [hunk 1]
```diff
@@ -43,6 +43,6 @@ resource "aws_nat_gateway" "nat_gws" {
   for_each = toset(var.private_subnet_cidr_blocks)
 
   allocation_id = aws_eip.nat_gw_eips[each.value].id
-  subnet_id     = module.private.subnets[each.value].id
+  subnet_id     = module.public.subnets[var.public_subnet_cidr_blocks[index(var.private_subnet_cidr_blocks, each.value)]].id
```
Comment (dav3r) [inline subnets.tf]: This line is now a bit of a brain-twister, at least for me- it took me a minute to grep what was going on.  Can you please add a comment here to assist future readers?

**Review by dav3r — APPROVED** [review]
Approved, but I requested one comment be added.  🏞

File: subnets.tf (hunk 1 above)
Comment (jsf9k) [reply subnets.tf]: This was something I intended to do but forgot about.  See commit 3d80193.

**Review by mcdonnnj — APPROVED** [review]
Nice work 👍

**Pull request merged**